### ISSUE TITLE
group articles of the Explanation section into proper subsections

### ISSUE THREAD

**Issue by jugmac00** [canonical/launchpad-manual#80]
see https://github.com/canonical/open-documentation-academy/issues/115

**Comment by nbarbarick** [comment]
Hi @jugmac00. As I’m trying to organize this section and create coherent subtopics, I’m wondering if all pages should be included in a reorganized Explanation section.

As of now, there are 18 topics under the General heading with a warning at the top of the page that the content might be outdated. 

Are any of the following pages obsolete or no longer needed?

* Launchpad Code Hosting
* Launchpad Mail 
* Launchpad Registry 
* About Malone 
* Feature Flags 
* How to go about writing a web application 
* Hacking 
* Engineering Overview: Translations 
* URL Traversal 
* Navigation menus 
* About Launchpad branches 
* Pre merge reviews 
* The Launchpad PPA 
* Testing 
* Datetime Usage Guide 
* Error explanations 
* Launchpad Security Policy
* XXX policy

**Comment by jugmac00** [comment]
Hi @nbarbarick,

Thanks for your continued support!

It is the exact right step to evaluate the content before starting to regroup the articles.

Last year when the Launchpad team migrated from the old wiki to readthedocs, we already sorted out articles which seem to not offer any value any more - mostly.

While basically all of the listed articles need a more or less significant revision, they are still very valid - with the exception of "How to go about writing a web application" - this can be removed for good. If in the same pull request, please put the removal in a separate commit.

The rest needs revision from the Launchpad team. We are currently growing and I am positive that we will be able to tackle more necessary documentation work over time.

I could already imagine a couple of subcategories for the remaining items.

- "The Launchpad PPA" is about setting up an environment.
- "About Malone" is about describing a part of the system
- "Testing", "XXX policy", "Hacking" and quite some more are general information for Launchpad engineers
- "Error explanations" might be a good fit for a debugging subsection.
- "URL traversal", "Navigation menus", "Launchpad mail" touch a very specific topic... not sure how to name that group.

Does that make sense?

Thanks again! Your support is highly appreciated!

**Comment by nbarbarick** [comment]
Hi @jugmac00.

Thanks for your suggestions. Here is an attempt to categorize the pages in the Explanation > General section. Once we have solidified the organization, I can suggest some introductory text for each section.

Please let me know of any changes you'd like to see. 

***

# General

## Understanding the Launchpad project

* What is Launchpad?
* Launchpad strategy
* Launchpad values

## Setting up and understanding Launchpad

* Launchpad installation details 
* Application framework
* Navigating the tree 
* Launchpad pip integration
* The Launchpad PPA

## Best practices for development

* Architectural guide
* Charm development
* Code import in depth
* About Launchpad performance
* About Launchpad security 
* Template reuse 

## How Launchpad works

* Launchpad Code Hosting
* Navigation menus
* Launchpad Mail
* Launchpad Registry
* URL Traversal

## Developing the Launchpad project

* About Malone
* About Launchpad branches
* Pre merge reviews
* Error explanations

## General information

* Hacking
* Engineering Overview: Translations
* Feature Flags
* Testing
* Datetime Usage Guide
* Launchpad Security Policy
* XXX policy

**Comment by jugmac00** [comment]
Hi @nbarbarick,

Unfortunately my browser crashed when my answer was almost finished, so here again. Thanks for your patience!

I think this looks already really good!

A couple of suggestions.

I would think that "About Malone", "Launchpad Registry", "Engineering Overview: Translations" and "Launchpad Code Hosting" would fit nicely together in a "Parts of Launchpad" subsection.

"Understanding the Launchpad project" and "Setting up and understanding Launchpad" sound a bit too similar - but I get your idea here.

What about renaming the first section into "The ideas behind Launchpad"?

I struggle a little bit with the "Best practices for development" and "General information" sections... they both seem to be about Launchpad development.

We could move "Testing" to the setup instructions.

Oh, it looks like we have two security items - we should make sure to have very distinctive titles.

I think "About Launchpad security" should go into the new section "ideas about Launchpad" as this is not so much about development practices. Whereas the second is.

So "Launchpad Security Policy" should go into "Best practices".

I think that is already quite a lot, and nothing is set in stone. I would be very happy if you could propose those changes including a decent introductionary paragraph. And in any case we think we could move an item around, we can do this anytime later.

But this seems like a major improvement!

Thanks for working on that!

**Comment by nbarbarick** [comment]
Hi @jugmac00,

Thanks for your feedback. I have drafted another version responsive to your notes, and did a bit more rearranging to hopefully make more coherent subsections. 

As far as the introductory text goes, I am trying to keep it general (and accurate) enough for all the included pages, and I welcome any suggestions you might have.

# General

## The ideas behind Launchpad

Explore these pages for an overview of the concepts and motivations behind the Launchpad project. 

* What is Launchpad?
* Launchpad strategy
* Launchpad values
* About Launchpad security 

## Setting up and understanding Launchpad

Understanding how to install and configure Launchpad, navigate the codebase, and test your changes will help you make meaningful contributions to the project. 

* Launchpad installation details 
* Application framework
* Navigating the tree 
* Launchpad pip integration
* The Launchpad PPA
* Testing

## Best practices for development

By using these principled approaches to development, you can improve Launchpad in a way that is sustainable and maintainable.

* Architectural guide
* Charm development
* Code import in depth
* About Launchpad performance
* Template reuse
* ~~Launchpad Security Policy~~ Launchpad permissions

## Developing the Launchpad project

These pages help explain important aspects of the developer experience.

* About Launchpad branches
* Pre merge reviews
* Error explanations
* Hacking
* Feature Flags
* Datetime Usage Guide
* XXX policy

## Parts of Launchpad

After understanding the goals and approaches of the project, dive into the details by exploring the below components of the application.

* Navigation menus
* Launchpad Mail
* URL Traversal
* About Malone
* Launchpad Registry
* Engineering Overview: Translations
* Launchpad Code Hosting

**Comment by jugmac00** [comment]
I love it! Please go ahead and propose this as is!

**Comment by alvaro-crespo** [comment]
This has been solved, thanks @nbarbarick !
